src/plugins/dashboard/server/dashboard_saved_object/dashboard_saved_object.ts

```diff
@@ -60,6 +90,7 @@ export const createDashboardSavedObjectType = ({
           controlStyle: { type: 'keyword', index: false, doc_values: false },
           chainingSystem: { type: 'keyword', index: false, doc_values: false },
           panelsJSON: { type: 'text', index: false },
+          showApplySelections: { type: 'boolean', index: false },
```

Comment:
When I try to remove these mappings, I get the following error when trying to launch Kibana:

```
Type dashboard: mappings added on model versions not present on the global mappings definition: controlGroupInput.properties.showApplySelections.type,controlGroupInput.properties.showApplySelections.index
```
Is there something I'm missing here? 🙈

Reply:
I see my original comment was a bit unclear :), I don't think we need to remove the other fields in this PR, just avoid introducing a new one for `showApplySelections` since we have a budget of how many fields we can map in ES.

---

I believe we also need to remove the operation to add a mapping from the `changes` key:

```diff
diff --git a/src/plugins/dashboard/server/dashboard_saved_object/dashboard_saved_object.ts b/src/plugins/dashboard/server/dashboard_saved_object/dashboard_saved_object.ts
index 79086b814f9..d4cbc204b3d 100644
--- a/src/plugins/dashboard/server/dashboard_saved_object/dashboard_saved_object.ts
+++ b/src/plugins/dashboard/server/dashboard_saved_object/dashboard_saved_object.ts
@@ -49,18 +49,7 @@ export const createDashboardSavedObjectType = ({
       },
     },
     2: {
-      changes: [
-        {
-          type: 'mappings_addition',
-          addedMappings: {
-            controlGroupInput: {
-              properties: {
-                showApplySelections: { type: 'boolean', index: false },
-              },
-            },
-          },
-        },
-      ],
+      changes: [],
       schemas: {
         forwardCompatibility: dashboardAttributesSchemaV2.extends({}, { unknowns: 'ignore' }),
         create: dashboardAttributesSchemaV2,
@@ -89,7 +78,6 @@ export const createDashboardSavedObjectType = ({
           controlStyle: { type: 'keyword', index: false, doc_values: false },
           chainingSystem: { type: 'keyword', index: false, doc_values: false },
           panelsJSON: { type: 'text', index: false },
-          showApplySelections: { type: 'boolean', index: false },
           ignoreParentSettingsJSON: { type: 'text', index: false },
         },
       },

```

Reply:
Ahh, good call, thank you - done in e048f77712fc146538f65c63dbdff3b79890c8bf :tada:

Reply:
@jloleysens Oh boy! That caused all sorts of explosions.... 

I got the following error for a bunch of functional tests when trying to import existing saved objects:
```
 │       error: {
 │         error: 'Internal Server Error',
 │         message: 'Unexpected bulk response [400] strict_dynamic_mapping_exception: [1:115] mapping set to strict, dynamic introduction of [showApplySelections] within [dashboard.controlGroupInput] is not allowed',
 │         statusCode: 500,
 │         type: 'unknown'
 │       },
```

🙈 I've had to add `showApplySelections` back to the mapping to resolve this... So sorry, but I must be missing something else! Any ideas? :eyes:

Reply:
I think (@jloleysens please keep me honest here 😅)  ideally the `dashboard.controlGroupInput` mapping in packages/kbn-check-mappings-update-cli/current_mappings.json would be similar to the [`lens.state` mapping](https://github.com/elastic/kibana/blob/main/packages/kbn-check-mappings-update-cli/current_mappings.json#L2020-L2028). The data is still stored in `_source` but not in separate fields. However, we might need multiple modelVersions to achieve that. 

e.g.
```
...
"dashboard": {
    "properties": {
      "controlGroupInput": {
        "dynamic": false,
        "properties": {}
      },
...
```

But I wonder if we can keep the `index: false, doc_values: false` settings to unblock this PR and remove those fields in a separate PR later?

Reply:
> But I wonder if we can keep the index: false, doc_values: false settings to unblock this PR and remove those fields in a separate PR later?

++ I just wanted to avoid introducing new mappings :). I don't want to block your progress on this, so I've approved, but missed your comment. Apologies for the delay. But I agree with your ideal end state 👍🏻 

> Oh boy! That caused all sorts of explosions....

Did you remove `showApplySelections` from your `mappings` object too?

Unfortunately the complete set of steps is a bit more complicated because we track new mappings as they come and there's also local ES state that'll need to be reset/wiped 😕 

The steps should be:

1. Apply the diff above to remove from `changes`
2. Remove `showApplySelections: { type: 'boolean', index: false, doc_values: false },` from the `mappings` in your SO declaration
3. Reset ES local state for your dev purposes
4. Run `node ./scripts/check_mappings_update.js --fix --no-verify` and commit any changes

You should have successfully removed the mapped field.

Reply:
> Did you remove showApplySelections from your mappings object too?

I did indeed, yes! I also updated the hash in `src/core/server/integration_tests/ci_checks/saved_objects/check_registered_types.test.ts` and updated the mapping in `packages/kbn-check-mappings-update-cli/current_mappings.json`. 

> The steps should be: <...>

I tried all of this, and I'm still seeing the same error when trying to run functional tests locally 😢 

```
 errors: [
 │     {
 │       error: {
 │         error: 'Internal Server Error',
 │         message: 'Unexpected bulk response [400] strict_dynamic_mapping_exception: [1:115] mapping set to strict, dynamic introduction of [showApplySelections] within [dashboard.controlGroupInput] is not allowed',
 │         statusCode: 500,
 │         type: 'unknown'
 │       },
 │       id: '3de3c3b0-6458-11ed-9957-e76caeeb9f75',
 │       meta: {
 │         icon: 'dashboardApp',
 │         title: 'timeslider and saved search'
 │       },
 │       type: 'dashboard'
 │     }
 │   ],
```

To avoid holding up this PR, let's do what @nickpeihl suggested and maybe clean up **all** dashboard saved object mappings in a separate PR :+1:

Reply:
++ I also see we'll want to migrate to `"dynamic": false,` that is the actual cause of the mappings error since we're running in `strict` mode. But happy for this to be addressed later!

Reply:
Created a follow up issue here: https://github.com/elastic/kibana/issues/178425 - thanks so much for your help @jloleysens :bow: